src/Battleships/Battleships.java

```diff
@@ -42,12 +44,37 @@ public int getCurrentPlayer() {
             return currentPlayer;
         }
 
+        public Player getPlayer(int index) {
+            return players.get(index);
+        }
+
         public void setCurrentPlayer(int currentPlayer) {
             this.currentPlayer = currentPlayer;
         }
 
-        public Player getPlayer(int index) {
-            return players.get(index);
+        public void setWinner(int player) {
+            winner = player;
+        }
+
+
+        public boolean hasWinner(Battleships battleships) {
+            for (int i = 0; i < 2; i++) {
+                int shipsleft = 5;
+                char[][] grid = battleships.getCurrentGameState().getPlayer(i).getGivenShots();
```

Comment:
Could you do it in a way that doesn't require you to pass battleships object in argument?
I am thinking that since you're asking for GameState (getCurrentGameState) and this code is already in GameState class then maybe it already has an access itself :P